src/utils/utils.tsx

```diff
@@ -0,0 +1,116 @@
+import { USER_ID } from '../api/todos';
+import { Setters } from '../types/Setters';
+import { Filter } from '../types/Filter';
+import { TodoWithLoader } from '../types/TodoWithLoader';
+import { handleDelete } from './handleDelete';
+import { handleUpdate } from './handleUpdate';
+import { State } from '../types/State';
+import { errorText } from '../constants';
+import { handleAdd } from './handleAdd';
+
+export const items = {
+  filter(todos: TodoWithLoader[], filter: Filter) {
+    switch (filter) {
+      case Filter.active:
+        return todos.filter(todo => !todo.completed);
+      case Filter.completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  },
+  completed(todos: TodoWithLoader[]) {
+    return todos.filter(todo => todo.completed);
+  },
+
+  uncompleted(todos: TodoWithLoader[]) {
+    return todos.filter(todo => !todo.completed);
+  },
+
+  clearCompleted(todos: TodoWithLoader[], setters: Setters) {
+    const completed = this.completed(todos);
+
+    completed.map(todo => {
+      handleDelete(todo, setters);
+    });
+  },
+
+  toggleAll(state: State, setters: Setters) {
+    const uncompleted = this.uncompleted(state.todos);
+
+    if (uncompleted.length > 0) {
+      uncompleted.map(todo => {
+        return handleUpdate(todo, !todo.completed, setters);
+      });
+    }
+
+    if (uncompleted.length === 0 && state.todos.length > 0) {
+      state.todos.map(todo => {
+        return handleUpdate(todo, !todo.completed, setters);
+      });
+    }
+  },
+};
+
+export const item = {
+  updateLoading(todo: TodoWithLoader, isLoading: boolean, setter: Setters) {
+    setter.setTodos(prevTodos => {
+      const index = prevTodos.findIndex(todo1 => todo1.id === todo.id);
+      const oldTodo = prevTodos[index];
+
+      if (index >= 0) {
+        setter.setUpdatedAt(new Date());
+        const newTodo: TodoWithLoader = { ...oldTodo, loading: isLoading };
+
+        prevTodos.splice(index, 1, newTodo);
+      }
+
+      return prevTodos;
```

Comment:
```suggestion
    setter.setTodos(prevTodos => 
     prevTodos.map(t => {
     if(t.id === todo.id) {
       setter.setUpdatedAt(new Date());
       return {...t, loading: isLoading}
     }
     
     return t;
     })
```
looks like we can simplify it